ros_ws/src/dataset_generation/include/dataset_generation/filter_point_clouds.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <dirent.h>
+#include <iostream>
+#include <thread>
+#include <string>
+#include <sstream>
+#include <unordered_set>
+
+#include <pcl/ModelCoefficients.h>
+#include <pcl/PCLPointCloud2.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/common/io.h>
+#include <pcl/point_types.h>
+#include <pcl/common/common_headers.h>
+#include <pcl/filters/voxel_grid.h>
+
+#include "extract_point_cloud_objects.h"
+
+#include <Eigen/Dense>
+#include <Eigen/Geometry>
+
+#include "dataset_generation/json.hpp"
+
+using namespace std::literals::chrono_literals;
+
+namespace dataset_generation
+{
+
+
+class PCDFiltering
+{
+public:
+    PCDFiltering(
+        std::string in_folder_path,
+        std::string object_name,
+        std::string out_folder_path);
```

Comment:
I don't think you need to pass these into the constructor. The only reason I had these in the Extraction class was because we wanted to count all the files and extract from them. Here we are only going to be doing it on file by file basis, so you should probably just have a default constructor `PCDFiltering()`

Reply:
I was thinking that if we can use the same path for all the methods that we are implementing in this class.